sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorAdministrationClient/NotificationHookLiveTests.cs

```diff
@@ -347,6 +353,69 @@ public async Task UpdateWebNotificationHookWithEveryMember()
             Assert.That(updatedWebHook.Headers, Is.EquivalentTo(headers));
         }
 
+        [RecordedTest]
+        [TestCaseSource(nameof(NotificationHookTestCases))]
+        public async Task UpdateCommonMembersWithNullSetsToDefault(string hookKind)
+        {
+            // https://github.com/Azure/azure-sdk-for-net/issues/21504
+            if (hookKind == nameof(NotificationHookKind.Webhook))
+            {
+                Assert.Ignore();
+            }
+
+            MetricsAdvisorAdministrationClient adminClient = GetMetricsAdvisorAdministrationClient();
+
+            string hookName = Recording.GenerateAlphaNumericId("hook");
+            NotificationHook hookToCreate = CreateMockNotificationHook(hookKind, hookName);
+
+            hookToCreate.Description = "description";
+            hookToCreate.ExternalUri = new Uri("https://fakeuri.com");
+
+            await using var disposableHook = await DisposableNotificationHook.CreateHookAsync(adminClient, hookToCreate);
+
+            NotificationHook hookToUpdate = disposableHook.Hook;
+
+            hookToUpdate.Description = null;
+            hookToUpdate.ExternalUri = null;
+
+            NotificationHook updatedHook = await adminClient.UpdateHookAsync(hookToUpdate);
+
+            Assert.That(updatedHook.Description, Is.Empty);
+            Assert.That(updatedHook.ExternalUri, Is.Null);
+        }
+
+        [RecordedTest]
+        [Ignore("https://github.com/Azure/azure-sdk-for-net/issues/21504")]
+        public async Task UpdateWebHookWithNullSetsToDefault()
+        {
+            MetricsAdvisorAdministrationClient adminClient = GetMetricsAdvisorAdministrationClient();
+
+            string hookName = Recording.GenerateAlphaNumericId("hook");
+            var hookToCreate = new WebNotificationHook(hookName, new Uri("https://fakeuri.com/"))
+            {
+                Username = "username",
+                Password = "password",
+                CertificateKey = "certKey",
+                CertificatePassword = "certPass"
+            };
+
+            await using var disposableHook = await DisposableNotificationHook.CreateHookAsync(adminClient, hookToCreate);
+
+            var hookToUpdate = disposableHook.Hook as WebNotificationHook;
+
+            hookToUpdate.Username = "username";
```

Comment:
maybe I just got confused with all the changes, but could you remind me what is the purpose here?
my understanding is that password-type things are not returned by the service, right? so here are u validating .. what?

Reply:
Refer to my comment in the other question. Only secrets in data sources are not returned.

Reply:
so should this be set to null? or why is ti that u assign a value and then the returned from the service is empty?

Reply:
It should be set to null. Fixed it.

Reply:
Ran it locally and noticed we have trouble validating the certification key, since the service does not seem to accept mock values (already tracked by https://github.com/Azure/azure-sdk-for-net/issues/17485). Also, I was not taking sanitization of passwords in account for tests. Fixed it.